src/main/scala/com/microsoft/hyperspace/actions/CreateActionBase.scala

```diff
@@ -181,6 +185,9 @@ private[actions] abstract class CreateActionBase(dataManager: IndexDataManager)
     val (resolvedIndexedColumns, resolvedIncludedColumns) = resolveConfig(df, indexConfig)
     val columnsFromIndexConfig = resolvedIndexedColumns ++ resolvedIncludedColumns
 
+    val escapedIndexedColumns = SchemaUtils.prefixNestedFieldNames(resolvedIndexedColumns)
```

Comment:
I just reviewed a comment of yours above: https://github.com/microsoft/hyperspace/pull/379#issuecomment-800554698 saying:

>Hmm, how about just not supporting nested columns if their names have "." in the first cut? We can throw an exception for now, and add it later if there is a demand.

So, the top level column even if it's nested it first level name should NOT contain `.` (dot). I guess we should also removed the condition of `> 1` in `ResolverUtils`.

Reply:
Yea, I think we should still support top-level columns with dots. For example, a top level column "\`x.y\`" should work where as a nested column "nested.\`x.y\`.z" should not be allowed. (thus the >1 check).

With the current code, wouldn't the top level column be also escaped? We can make `resolve` to return a tuple (resolved name, isNestedColumn) and call `prefixNestedFieldNames` only for nested columns.

I think the current hyperspace doesn't support top level column with dots, so we can address this as a separate PR; but let's make sure we are not prefixing top level column with dots.

Reply:
@imback82 Im not sure what you want. First you say that current Hyperspace does NOT support top level column with dots and then you are saying that we should still implement this extra work for a possible future feature that is not yet there. Why not doing it the other way around? I mean why not when adding this support into this code when the feature will be implemented?

I see 2 ways:
1. Keeping the current implementation of Hyperspace, the one that does NOT support dots and change the `> 1` so that it will throw the `HyperspaceException` when dots are encountered.
2. Implement the feature and change the `revolve` methods in `ResolverUtils` so that it supports the tuple output which will result in changes in other parts of the code.

I'll try to implement it but I hope that we will go further on with this PR because until now it did take a lot more than I expected and there are 4 or 5 more PRs coming to have this nested support in Hyperspace.

Reply:
No, I was suggesting to do it as a separate PR ("so we can address this as a separate PR"). I thought my ask (make `resolve` to return extra info and wrap it with `col`) was a simple change, but it's OK, I will do a follow up to clean this up.

For this PR, we still need to do the following:
1. Instead of removing `>1`, add:
```scala
if (resolvedColNameParts.length == 1 && resolvedColNameParts.head.contains(".")) {
  throw HyperspaceException(
    s"Hyperspace does not support the top-level column whose name contains dots: $origCol")
}
```
2. Clearly document the prerequisite for `prefixNestedFieldName`, since it depends on the check in `resolve`.